pkg/resolver/request.go

```diff
@@ -43,6 +44,18 @@ func RequestGETWithHeaders(url string, extraHeaders map[string]string) (response
 	return httpClient.Do(req)
 }
 
+func RequestPOST(url, body string) (response *http.Response, err error) {
+	req, err := http.NewRequest("POST", url, strings.NewReader(body))
```

Comment:
I was reading the [documentation](https://golang.org/pkg/net/http/#NewRequestWithContext) but I'm still not 100% sure. The documentation states:

> If the provided body is also an io.Closer, the returned Request.Body is set to body and will be closed by the Client methods Do, Post, and PostForm, and Transport.RoundTrip.

According to the [`strings.Reader` documentation](https://golang.org/pkg/strings/#Reader), it does not implement the `io.Closer` interface. So do we manually have to close the body?

Reply:
If it does not implement the `io.Closer` interface it means it does not need to be closed